src/bloqade/squin/wire.py

```diff
@@ -95,21 +95,24 @@ def __init__(self, operator: ir.SSAValue, *args: ir.SSAValue):
 class Measure(ir.Statement):
     traits = frozenset({lowering.FromPythonCall(), WireTerminator()})
     wire: ir.SSAValue = info.argument(WireType)
+    qubit: ir.SSAValue = info.argument(QubitType)
     result: ir.ResultValue = info.result(types.Int)
 
 
 @statement(dialect=dialect)
-class MeasureAndReset(ir.Statement):
-    traits = frozenset({lowering.FromPythonCall(), WireTerminator()})
-    wire: ir.SSAValue = info.argument(WireType)
+class NonDestructiveMeasure(ir.Statement):
+    traits = frozenset({lowering.FromPythonCall()})
+    input_wire: ir.SSAValue = info.argument(WireType)
     result: ir.ResultValue = info.result(types.Int)
     out_wire: ir.ResultValue = info.result(WireType)
 
 
 @statement(dialect=dialect)
-class Reset(ir.Statement):
+class MeasureAndReset(ir.Statement):
     traits = frozenset({lowering.FromPythonCall(), WireTerminator()})
     wire: ir.SSAValue = info.argument(WireType)
+    result: ir.ResultValue = info.result(types.Int)
+    out_wire: ir.ResultValue = info.result(WireType)
```

Comment:
wait I thought in the issue this was supposed to be a terminator? that means it should take in.a qubit and only return the result of the measurement.

Reply:
I'll be honest I'm a bit confused myself, what you're describing sounds like Measurement in the qubit dialect. All I really have from my notes is that MeasureAndReset was supposed to be removed from squin.qubit.

I also realize it's a bit confusing that the RFC conclusion was to remove MeasureAndReset from qubit dialect but right before that conclusion I say it should be removed from the wire dialect 😅